File: code/bit_manipulation/src/thrice_unique_number/thrice_unique_number.cpp
/*
 *
 * Part of Cosmos by OpenGenus Foundation
 * Find unique number in an array where every element occours thrice except one.Find that unique Number
 *
 */
#include <iostream>
using namespace std;
int n;
int a[104] = {1, 1, 1, 3, 3, 2, 3};
int f()
{
    int count[65] = {0};
    for (int i = 0; i < n; i++)
    {
        int j = 0;
        int temp = a[i];
        while (temp)
        {
            count[j] += (temp & 1);
            j++;
            temp = temp >> 1;
        }
    }
    int p = 1, ans = 0;
    for (int i = 0; i <= 64; i++)
    {
        count[i] %= 3;
        ans += (count[i] * p);
        p = p << 1;
    }
    return ans;
}
int main()
{
    n = sizeof(a) / sizeof(a[0]);
    cout << f();
    return 0;
}

File: code/bit_manipulation/src/thrice_unique_number/thrice_unique_number.js
// Part of Cosmos by OpenGenus Foundation

let numbers = [
  1,
  1,
  1,
  2,
  2,
  2,
  3,
  3,
  3,
  4,
  5,
  5,
  5,
  6,
  6,
  6,
  7,
  7,
  7,
  8,
  8,
  8,
  9,
  9,
  9
];
let counter = {};

for (var index = 0; index < numbers.length; index++) {
  let current = counter[numbers[index]];

  if (current === undefined) {
    counter[numbers[index]] = 1;
  } else {
    counter[numbers[index]] = current + 1;
  }
}

for (let num in counter) {
  let count = counter[num];

  if (count === 1) {
    console.log(`Unique number is ${num}`);
  }
}
